### PR TITLE
Use wpilib JDK if it exists

### LINKED ISSUE #53: Use WPILib Java Install
There have been multiple issues in the FRC discord with the program failing to deploy because it is using the global (therefore wrong) version of Java. 

This tool should use the JDK that is installed with WPILib as opposed to the global version of Java, installed at `WPILib/xxxx/jdk`. It should preferably be configurable by default to WPILib's included version.

### PR DESCRIPTION
**Pull request by pietroglyph**
Sometimes users have older JDKs installed and we end up using those and failing the deploy as a result. This PR will use the newest wpilib JDK if one is installed, and warn if one is not installed or if the current wpilib JDK does not match the current year.

This has been tested on Windows and Linux.

Closes #53.

### CODE REVIEW

File: frc_characterization/newproject/__init__.py [hunk 1]
```diff
@@ -114,18 +115,38 @@ def deployProject():
...
+                    "gradlew.bat",
+                ),
+                cmd,
+                "--console=plain",
+            ]
+
+            # This path is correct *as of* wpilib 2020
+            # Prior to 2020 the path was C:/Users/Public/frcYEAR/jdk
+            jdk_path = os.path.join(
+                "C:", os.sep, "Users", "Public", "wpilib", str(datetime.now().year), "jdk"
```
Comment (auscompgeek) [inline frc_characterization/newproject/__init__.py]: This is wrong for Windows installs not on C:\.

Comment (pietroglyph) [reply frc_characterization/newproject/__init__.py]: Fixed.

**Comment by pietroglyph** [comment]
@Oblarg This should be ready for review. 

Note that this is not compatible with pre-2020 wpilib paths (where all JDKs are under `~/frcYEAR/jdk`). I figured that just supporting 2020+ would be easier then remembering to make another PR in January. If you want us to support pre-2020 style wpilib paths for the next two months just let me know.

**Comment by pietroglyph** [comment]
@sciencewiz This PR does not use the current date for version detection. It only uses the current date to warn the user that they *may* have an out of date JDK.

**Comment by auscompgeek** [comment]
This should let the user pick the JDK from the options, and maybe even not do anything if $JAVA_HOME is set.

**Comment by pietroglyph** [comment]
@auscompgeek I can add options, but I do not think that we should use the default JDK just if Java home is set: most of the issues that users have had have been as a result of having a JAVA_HOME that points to an old JDK.

**Comment by Daltz333** [comment]
I second that. It should use the wpilib installed if it's available.

**Comment by Oblarg** [comment]
It should look for the current year's WPILib installation (in this case, 2020) and, if it doesn't find that, use JAVA_HOME.

**Pull request merged**